meilisearch-lib/src/index/updates.rs

```diff
@@ -323,19 +323,34 @@ impl Index {
             indexing_callback,
         )?;
 
+        let mut results = Vec::new();
         for content_uuid in contents.into_iter() {
             let content_file = file_store.get_update(content_uuid)?;
-            let reader = DocumentBatchReader::from_reader(content_file)?;
-            builder.add_documents(reader)?;
-        }
-
-        let addition = builder.execute()?;
+            let reader = DocumentsBatchReader::from_reader(content_file)?;
+            let (new_builder, user_result) = builder.add_documents(reader)?;
+            builder = new_builder;
+
+            let user_result = match user_result {
+                Ok(count) => {
+                    let addition = DocumentAdditionResult {
+                        indexed_documents: count,
+                        number_of_documents: count,
+                    };
+                    info!("document addition done: {:?}", addition);
+                    Ok(addition)
+                }
+                Err(e) => Err(IndexError::from(e)),
+            };
 
-        txn.commit()?;
+            results.push(user_result);
+        }
 
-        info!("document addition done: {:?}", addition);
+        if results.iter().any(Result::is_ok) {
+            let _addition = builder.execute()?;
+            txn.commit()?;
+        }
```

Comment:
This is where everything happens, and this little if at the end is actually really important.
Maybe we should add some tests to ensure we never forget how useful it is?
- Push 3 updates that all fails and ensure nothing happened
- Push 2 failed updates + 1 successful update and ensure it really processed while the two other got an error